src/SonarScanner.MSBuild.Common/AnalysisConfig/AnalyzerPlugin.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Xml.Serialization;
+
+namespace SonarScanner.MSBuild.Common
+{
+    /// <summary>
+    /// XML-serializable data class for a single SonarQube plugin containing an analyzer
```

Comment:
New data class. This is effectively a copy of the existing _Plugin_ class, with the addition of the _AssemblyPaths_ property.

The existing _Plugin_ class is only in the pre-processor to store the results of a call to the server. Creating a new class to be returned by the _RoslynAnalyzerProvider_ seemed cleaner than overloading the existing class.

Reply:
The `Plugin` class is going to be replaced with an equivalent from the Client.